### PR TITLE
Install rules cleanup

### LINKED ISSUE #220: Clean install rules up?
Considering my last sentence in https://github.com/Dobiasd/FunctionalPlus/pull/216#issue-547935166 I feel like that it should be mentioned that many things in the `install-rules.cmake` are remnants of a (not that good) template and I feel like the file should be reworked a bit.

I would remove some comments and variables that aren't really essential, inline the `${PROJECT_NAME}` dereferences (how often does a project change its name?), make code a little tighter, but still preserve the information in the form of comments to stay in the spirit of what I said in that PR.

This issue is to survey whether this change is desired or not.

### PR DESCRIPTION
**Pull request by friendlyanon**
Fixed #220

Please take a look at the comments and let me know if you think they are informative enough without being redundant.

### CODE REVIEW

**Comment by Dobiasd** [comment]
My CMake-fu is way too unsophisticated to review in a meaningful way.

@offa Would you like to have a look? :grin:

**Comment by offa** [comment]
Already :+1:  :-)

File: cmake/install-rules.cmake [hunk 1]
```diff
@@ -1,72 +1,50 @@
...
 )
 
-# Config
-#   * ${CMAKE_INSTALL_LIBDIR}/cmake/FunctionalPlus/FunctionalPlusConfig.cmake
-#   * ${CMAKE_INSTALL_LIBDIR}/cmake/FunctionalPlus/FunctionalPlusConfigVersion.cmake
+# Use the PROJECT_VERSION variable to generate a basic config version file in
+# the PROJECT_BINARY_DIR directory
+write_basic_package_version_file(
+    FunctionalPlusConfigVersion.cmake
+    COMPATIBILITY SameMajorVersion
```
Comment (offa) [inline cmake/install-rules.cmake:27]: Maybe we can even use the minor here? @Dobiasd all versions 0.2.x are more or less compatible?

Comment (friendlyanon) [reply cmake/install-rules.cmake:27]: Is there a need to steer away from semver?

Comment (Dobiasd) [reply cmake/install-rules.cmake:27]: > @Dobiasd all versions 0.2.x are more or less compatible?

Yeah, they should be. I just went through the changelogs, and I think it was all:
- bug fixes
- new functionality
- performance improvements
- docs
- build system

I can't guarantee 100% backward compatibility, but there were no things breaking it intentionally. So it'd bluntly answer with "Yes.". :slightly_smiling_face:

Comment (friendlyanon) [reply cmake/install-rules.cmake:27]: For the record, here is the documentation for [write_basic_package_version_file](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file) that explains what the `COMPATIBILITY` argument means.

Comment (offa) [reply cmake/install-rules.cmake:27]: > Is there a need to steer away from semver?

No, just an idea. It shouldn't block the PR btw.

Comment (Dobiasd) [reply cmake/install-rules.cmake:27]: Shall we resolve this conversation and merge the PR?

Comment (offa) [reply cmake/install-rules.cmake:27]: Yes, please.

**Pull request merged**